### DISCUSSION TITLE
Cannot install Ganache

### DISCUSSION THREAD

**Discussion by taigaisatiger** [ConsenSys-archive/ganache#2672] [Q&A]
Hello I have a problem where I am not able to install Ganache

node --verison is v16.14.2.
npm --version is 8.5.0.
nvm --version is 0.39.1.
I am using MacOS version 11.6.5.
Visual Studio Code version is 1.63.2.

When I run `npm i -g ganache-cli`, I get the response below.
I am not really sure what the problem is here, so could anyone help me?

```
npm WARN deprecated ganache-cli@6.12.2: ganache-cli is now ganache; visit https://trfl.io/g7 for details

changed 6 packages, and audited 102 packages in 4s

2 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (7 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

**Comment by davidmurdoch** [comment]
`ganache-cli` is now `ganache`; visit https://trfl.io/g7 for details :-)


If you don't want to visit the link above you can try running `npm uninstall ganache-cli -g && npm install ganache -g`.

Ganache v7 is a new major version and contains some changes that may be breaking for some users. Also, the package was renamed from ganache-cli to just ganache.

**Reply by taigaisatiger** [reply]
Thank you for your answer
However, I got this error message.



```
removed 101 packages, and audited 1 package in 1s

found 0 vulnerabilities
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /Users/taiga/.npm-global/lib/node_modules/ganache
npm ERR! dest /Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u
npm ERR! errno -66
npm ERR! ENOTEMPTY: directory not empty, rename '/Users/taiga/.npm-global/lib/node_modules/ganache' -> '/Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u'

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taiga/.npm/_logs/2022-03-22T01_05_28_430Z-debug-0.log
```

**Reply by taigaisatiger** [reply]
When I try to uninstall ganache or ganache-cli, I get similar errors and am not able to uninstall as well, so I am having a problem.

**Reply by davidmurdoch** [reply]
Try removing all references to ganache in the above folders first.

**Reply by taigaisatiger** [reply]
Am I supposed to delete the files running the command below?
`rm -rf ganache`

**Reply by davidmurdoch** [reply]
Delete them from within this location: /Users/taiga/.npm-global/lib/node_modules/

**Reply by taigaisatiger** [reply]
So I am supposed to run this?

`rm -rf /Users/taiga/.npm-global/lib/node_modules/`

**Reply by taigaisatiger** [reply]
I tried it, but when I $ls in the command, it looks like it is not deleted.
Maybe it is because of Brownie(https://eth-brownie.readthedocs.io/en/stable/) that I am running?

**Reply by taigaisatiger** [reply]
Under node_modules, there is nothing in there

**Reply by taigaisatiger** [reply]
So I ran this code `npm uninstall ganache-cli -g && npm install ganache -g`,  but it is an error again. Same one as before.

```
up to date, audited 1 package in 648ms

found 0 vulnerabilities
npm ERR! code ENOTEMPTY
npm ERR! syscall rename
npm ERR! path /Users/taiga/.npm-global/lib/node_modules/ganache
npm ERR! dest /Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u
npm ERR! errno -66
npm ERR! ENOTEMPTY: directory not empty, rename '/Users/taiga/.npm-global/lib/node_modules/ganache' -> '/Users/taiga/.npm-global/lib/node_modules/.ganache-ZmLEla8u'

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/taiga/.npm/_logs/2022-03-22T02_39_41_844Z-debug-0.log
```

**Reply by davidmurdoch** [reply]
Try `npm uninstall ganache-cli -g && npm uninstall ganache -g && npm install ganache -g`?

I don't think brownie is a problem here.

**Reply by taigaisatiger** [reply]
I get the response below when I run that.

```

up to date, audited 1 package in 626ms

found 0 vulnerabilities

removed 29 packages, and audited 1 package in 518ms

found 0 vulnerabilities

added 29 packages, and audited 30 packages in 1m

found 0 vulnerabilities
```

However, when I run `ganache` or `ganache --version`, I get this.
`zsh: command not found: ganache`

**Reply by davidmurdoch** [reply]
Are you able to globally install and use any other npm packages? Try something like `npm install nodemon -g && nodemon -v`?

**Reply by taigaisatiger** [reply]
I cannot globally install nodemon. I tried and I get the same error `zsh: command not found: nodemon`. Looks like installing globally is a problem?

**Reply by davidmurdoch** [reply]
Looks like your node installation isn't set up quite right.

Maybe try the Solution here?https://medium.com/@jagatjyoti.1si13cs040/npm-g-install-npm-package-not-working-as-desired-why-why-why-19795abf0b59

Whatever you do, don't try installing an npm package with sudo (if you already have you should really consider formatting your drives and starting over).

**Reply by taigaisatiger** [reply]
This is what I get when I run the codes. When I run `ganache` on a different terminal in VScode, I get `zsh: command not found: ganache`. 

```
taiga@taiganoMacBook-Pro brownie_simple_storage % npm list -g --depth=0
/Users/taiga/.npm-global/lib
├── ganache@7.0.3
├── lite-server@2.6.1
├── nodemon@2.0.15
└── typescript@4.6.2

taiga@taiganoMacBook-Pro brownie_simple_storage % npm bin -g
/Users/taiga/.npm-global/bin
(not in PATH env variable)
taiga@taiganoMacBook-Pro brownie_simple_storage % mkdir ~/.npm-global
mkdir: /Users/taiga/.npm-global: File exists
taiga@taiganoMacBook-Pro brownie_simple_storage % npm config set prefix `~/.npm-global`
zsh: permission denied: /Users/taiga/.npm-global
taiga@taiganoMacBook-Pro brownie_simple_storage % export PATH=~/.npm-global/bin:$PATH
taiga@taiganoMacBook-Pro brownie_simple_storage % source ~/.profile
source: no such file or directory: /Users/taiga/.profile
taiga@taiganoMacBook-Pro brownie_simple_storage % npm install -g ganache
⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu⸨##################⸩ ⠏ reify:leveldown: timing reifyNode:node_modules/ganache/node_modu
added 29 packages, and audited 30 packages in 1m

found 0 vulnerabilities
taiga@taiganoMacBook-Pro brownie_simple_storage % ganache
ganache v7.0.3 (@ganache/cli: 0.1.4, @ganache/core: 0.1.4)
Starting RPC server
Error: listen EADDRINUSE: address already in use 127.0.0.1:8545.
    at /Users/taiga/.npm-global/lib/node_modules/ganache/dist/node/1.js:2:205165
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
taiga@taiganoMacBook-Pro brownie_simple_storage % ganache --version
ganache v7.0.3 (@ganache/cli: 0.1.4, @ganache/core: 0.1.4)
```

**Reply by taigaisatiger** [reply]
`Error: listen EADDRINUSE: address already in use`
Since I had this error, I ran `kill -9 <process_id> `because I saw this solution. https://ethereum.stackexchange.com/questions/37314/error-listen-eaddrinuse-8545#:~:text=This%20simply%20means%20you%20have,and%20if%20so%2C%20kill%20it.

**Reply by taigaisatiger** [reply]
But I still get `zsh: command not found: ganache` when I run `ganache` on a different terminal inside VScode

**Reply by davidmurdoch** [reply]
Sorry I don't know much about zsh. Maybe search for something like "node global module and zsh" to try to find a solution.

**Reply by taigaisatiger** [reply]
Ok, thank you.

If I were to restart this whole thing like you said here,
`Whatever you do, don't try installing an npm package with sudo (if you already have you should really consider formatting your drives and starting over).`

what should I do? And is there any way to check if I had run sudo before?
